src/interface/metadata.hpp

```diff
@@ -217,6 +233,20 @@ class Metadata {
     return None;
   }
 
+  MetadataFlag Dependency() const {
```

Comment:
can we change the name of this?  not sure to what.  to me this name suggests that it provides some means of having the current variable depend on other variables.

Reply:
How about something like `Role`?